File: src/test/java/com/hyak4j/PersonTest.java
package com.hyak4j;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class PersonTest {

    @Test
    public void bmiTest(){
        final float weight = 66.5f;
        final float height = 1.7f;
        Person person = new Person(weight, height);
        Assertions.assertEquals( weight /(height * height) , person.bmi());
    }
}

File: src/test/kotlin/com/hyak4j/PersonTest.kt
package com.hyak4j.kotlin

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class PersonTest {

    // 測試Kotlin類別
    @Test
    fun humanBmiTest(){
        val human = Human(66.5f, 1.7f)
        Assertions.assertEquals(66.5f / (1.7f * 1.7f), human.bmi())
    }

    // 測試JAVA類別
    @Test
    fun personBmiTest(){
        val person = Person(66.5f, 1.7f)
        Assertions.assertEquals(66.5f / (1.7f * 1.7f), person.bmi())
    }
}